components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/spark/algorithms/SparkModelUtils.java

```diff
@@ -421,4 +421,19 @@ public Boolean call(Tuple2<Double, Double> pl) {
         
         return uniqueValues;
     }
+
+    /**
+     * Utility method to get key from value of a map
+     *
+     * @param map       Map to be searched for a key
+     * @param value     Value of the key
+     */
+    public static <T, E> T getKeyByValue(Map<T, E> map, E value) {
+        for (Map.Entry<T, E> entry : map.entrySet()) {
+            if (Objects.equals(value, entry.getValue())) {
+                return entry.getKey();
+            }
+        }
+        return null;
+    }
```

Comment:
@cdathuraliya can we move this to org.wso2.carbon.ml.core.utils.MLUtils since this is not related to Spark?